apps/condo/domains/common/utils/pdf.ts

```diff
@@ -81,3 +83,28 @@ export const createPdf: ICreatePdf = (options) => {
         return doc.save(fileName, { returnPromise: true })
     })
 }
+
+export const createPdfWithPageBreaks: ICreatePdfWithPageBreaks = (options) => {
+    const { element, fileName } = options
+    const elementWidth = element.clientWidth
+    const elementHeight = element.clientHeight
+    const topLeftMargin = 15
+    const pdfWidth = elementWidth + (topLeftMargin * 2)
+    const pdfHeight = (pdfWidth * 1.5) + (topLeftMargin * 2)
+
+    const totalPDFPages = Math.ceil(elementHeight / pdfHeight) - 1
+
+    return html2canvas(element, { allowTaint: true }).then(canvas => {
+        const imgData = canvas.toDataURL('image/jpeg', 1.0)
+        const pdf = new Jspdf('p', 'pt',  [pdfWidth, pdfHeight])
```

Comment:
get pdfWIdth, pdfHeight from settings

Reply:
I can't get pdfWIdth, pdfHeight params from constant because thats related to problem with chart display. In this case document properties should be calculated from element rendered before pdf save.